woocommerce/woocommerce
# update api-core-tests to work on daily site and localhost/CI

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Allows api-core-tests to run against the daily test api site.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. create env locally with `pnpn env:test`
2. navigate to `/plugins/woocommerce` and execute the 2 updated tests using:
3. `pnpm test:api-pw system-status/system-status-crud.test.js` and  
4. `pnpm test:api-pw payment-gateways/payment-gateways-crud.test.js`
5. run all api tests locally with `pnpm test:api-pw`
6. Run the tests against the daily test env by creating a `.env` file in the `api-core-tests` directory with the appropriate values from the [smoke-test-daily.yml](https://github.com/woocommerce/woocommerce/blob/trunk/.github/workflows/smoke-test-daily.yml) workflow in the  api-tests job for:
API_BASE_URL=${{ secrets.SMOKE_TEST_API_URL }}
USER_KEY=${{ secrets.SMOKE_TEST_ADMIN_USER }}
USER_SECRET=${{ secrets.SMOKE_TEST_ADMIN_PASSWORD }}
UPDATE_WC: 'nightly'
7. repeat tests in steps 3, 4 and 5
8. Ensure CI tests pass



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
